syntax = "proto3";

package asterix;

service Asterix{
  // A simple RPC.
  // Obtains the MessageResponse at a given position.

 rpc GetServerResponse(Message) returns (MessageResponse) {}
 
 rpc Lookup(Stockname) returns (Stockprice) {}

 rpc Trade(TradeInfo) returns (Success) {}

 rpc Update(UpdateInfo) returns (Success) {}
}


message Message{
  string message = 1;
}

message MessageResponse{
  string message = 1;
  bool received = 2;
}

message Stockname{
  string stockName = 1;
}

message Stockprice{
  float price = 1;
}

message TradeInfo{
  string stockName = 1;
  int32 N = 2;
  string type =3;
}

message UpdateInfo{
  string stockName = 1;
  float price = 2;
}

message Success{
  int32 response = 1;
}